tapchannel/commitment.go

```diff
@@ -730,6 +730,13 @@ func CreateAllocations(chanState lnwallet.AuxChanState, ourBalance,
 				"allocation, HTLC is dust")
 		}
 
+		// To ensure uniqueness of the script key across HTLCs with the
+		// same payment hash and timeout (which would be equal
+		// otherwise), we tweak the asset level internal key of the
+		// script key with the HTLC index. We'll ONLY use this for the
+		// asset level, NOT for the BTC level.
+		tweakedTree := TweakHtlcTree(htlcTree, htlc.HtlcIndex)
```

Comment:
The other relevant area here is in the aux sweeper, when we go to derive the control block and witness scripts.

Reply:
Yes. I've now based this PR on your sweeper PR and added that code.